python-checks/src/main/java/org/sonar/python/checks/tests/TestShouldBeSkippedExplicitlyCheck.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.tests;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.tree.AssertStatement;
+import org.sonar.plugins.python.api.tree.AssignmentStatement;
+import org.sonar.plugins.python.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.FunctionDef;
+import org.sonar.plugins.python.api.tree.IfStatement;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.ReturnStatement;
+import org.sonar.plugins.python.api.tree.Statement;
+import org.sonar.plugins.python.api.tree.Tree;
+
+@Rule(key = "S5918")
+public class TestShouldBeSkippedExplicitlyCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Skip this test explicitly.";
+  private static final Set<String> supportedAssertMethods = new HashSet<>(Arrays.asList("assertEqual", "assertNotEqual", "assertTrue",
+    "assertFalse", "assertIs", "assertIsNot", "assertIsNone", "assertIsNotNone", "assertIn", "assertNotIn", "assertIsInstance",
+    "assertNotIsInstance", "assertRaises", "assertRaisesRegexp", "assertRaisesRegex", "assertAlmostEqual", "assertNotAlmostEqual",
+    "assertGreater", "assertGreaterEqual", "assertLess", "assertLessEqual", "assertRegexpMatches", "assertNotRegexpMatches",
+    "assertItemsEqual", "assertDictContainsSubset"));
+  private static final Tree.Kind[] literalsKind = {Tree.Kind.STRING_LITERAL, Tree.Kind.NUMERIC_LITERAL, Tree.Kind.LIST_LITERAL,
+    Tree.Kind.BOOLEAN_LITERAL_PATTERN, Tree.Kind.NUMERIC_LITERAL_PATTERN, Tree.Kind.NONE_LITERAL_PATTERN, Tree.Kind.STRING_LITERAL_PATTERN,
+    Tree.Kind.SET_LITERAL, Tree.Kind.DICTIONARY_LITERAL};
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, ctx -> {
```

Comment:
Very good point ! I believe we discussed this with @guillaume-dequenne-sonarsource 
What do you think ? Should I go forward and add this check ?

Reply:
I think this makes sense when checking `unittest` cases, because you always have to import unittest (and be in a class that inherits from `unittest.TestCase`) for a test to be discoverable (from the top of my head, specifics to be verified).
For `pytest`, I believe that you don't need to import it. When `pytest` will run, it will discover anything that starts with `test` as a test, without necessarily being imported. So I'd definitely not check this one.

For `unittest`, I'd be fine not checking it either (though it may be more correct to do so), but in that case I'd probably document the potential FP (calling `self.assertX` in a class that is not a `unittest.TestCase` for example).

Reply:
I ended up adding the check of unittest presence.
It took more time than expected, because :
- sometimes unittest framework is wrapped around another framework, so the FQN is not necessary "unittest.case.TestCase" but can be something like "twisted.trial.unittest.TestCase". After a discussion with @andrea-guarino-sonarsource, we switched to a different approach : ensure that the FQN both contains "unittest" and "TestCase"
- they can be several level of superclass, so we have to add in the scope parent classes recursively

It is interesting to note that we get far fewer issue from PythonRulingTest : this is because the project tensoerflow have apparently they own test classes definitions (although it look a lot like unittest).